File: app/src/main/java/com/modori/kwonkiseokee/AUto/DevicePhotos/DevicePhotoDao.java
package com.modori.kwonkiseokee.AUto.DevicePhotos;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;


import com.modori.kwonkiseokee.AUto.data.DevicePhotoDTO;

import java.util.List;

@Dao
public interface DevicePhotoDao {
    @Insert()
    void insert(DevicePhotoDTO devicePhoto);

    @Delete()
    void deletePhotobyId(String photoId);

    @Query("SELECT count(*) FROM tag_table")
    int getPhotosCount();

    @Query("SELECT * from device_photos")
    LiveData<List<DevicePhotoDTO>> getPhotoLists();
}

File: app/src/main/java/com/modori/kwonkiseokee/AUto/data/DevicePhotoDao.kt
package com.modori.kwonkiseokee.AUto.data

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query

@Dao
interface DevicePhotoDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(devicePhotoDTO: DevicePhoto)

    @Query("DELETE FROM device_photos WHERE photoId =:photoId")
    fun deletePhotoById(photoId: String)

    @Query("SELECT count(*) FROM device_photos")
    fun getPhotosCount(): Int

    @Query("SELECT * FROM device_photos")
    fun getDevicePhotoList(): LiveData<List<DevicePhoto>>
}